torchbenchmark/score/compute_score.py

```diff
@@ -7,148 +7,151 @@
+            spec = yaml.full_load(spec_file)
+
+        self.weights = defaultdict(float)
+        category_spec = spec['hierarchy']['model']
+        domain_weight = 1.0/ len(category_spec)
+        for domain in category_spec:
+            tasks = category_spec[domain]
+            task_weight = 1.0 / len(tasks)
+            for task in tasks:
+                benchmarks = tasks[task]
+                benchmark_weight = 1.0 / len(benchmarks)
+                self.weights[task] = domain_weight * task_weight * benchmark_weight
+
+    def setup_benchmark_norms(self, data):
+        """
+        Helper function which gets the normalization values per benchmark
+        by going through the reference data file.
+
+        This reference data can be user provided or by default the first file
+        in the directory of data files is considered as the reference data file
+        """
+        self.norm = {b['name']: b['stats']['mean'] for b in data['benchmarks']}
+
+    def _get_model_task(self, model_name):
+        """
+        Helper function which extracts the task the model belongs to
+        by iterating over the Model attributes.
+        """
+        MODEL_PATH="../../models"
+        p = Path(__file__).parent.joinpath(MODEL_PATH + model_name)
+        if(p):
+            module = importlib.import_module(f'torchbenchmark.models.{model_name}', package=__name__)
+            Model = getattr(module, 'Model')
+        return Model.task.value
+
+    def _get_benchmark_configuration(self, b):
+        """
+        This helper function extracts the configuration details
+        from the benchmark name.
+        For eg., if benchmark_name = test_eval[pytorch_struct-cuda-jit]
+                test, device, mode = eval, cuda, jit
+                model_name = pytorch_struct
+
+        This helper also extracts the task and model_name of the benchmark
+        The function returns a tuple of the configs, model task, model_name
+        """
+        test, device, mode = "train", "cuda", "eager"
+        if "eval" in b: test = "eval"
+        if "cpu" in b: device = "cpu"
+        if "jit" in b: mode = "jit"
+
+        # Extract the model name from the benchmark b
+        model_name = (re.findall(r'\[(.*)\-'+device, b))[0]
+
+        # Get the Model task value by reading the Model attributes.
+        task = self._get_model_task(model_name)
+        return (test, device, mode), task, model_name
+
+    def _construct_benchmark_run_db(self, data):
+        """
+        Construct a benchmark database by going over through the data file
+        for the run and update the dictionary by task and model_name
+
+        For eg., the (key, value) for this dictionary is of the form
+        [generation][pytorch_stargan] = [(1.2, (eval, cuda, jit),
+                                        test_eval[pytorch_stargan-cuda-jit])]
+        """
+        found_benchmarks = defaultdict(lambda: defaultdict(list))
+
+        for b in data['benchmarks']:
```

Comment:
wondering if it makes sense to split this out as a helper function.  

it seems like it might be cleaner to just combine this loop with the loops in get_score_per_config, since there doesn't seem to be any value in having the 'db' object exist or keeping it around, and adding the helper just forces you to maintain 2 functions and comments for them, making it more burden to read the code.

Reply:
if you feel it's easier to construct a data structure and then use it to guide the iterator for the score computation, wouldn't it be better to construct the data structure using the ref_data rather than the computation data?

Reply:
I have moved this dictionary construction in `get_score_per_config`.